### PR TITLE
Address Receipts

### PR DESCRIPTION
**Pull request by OR13**
This PR makes it clearer that merkle trees are not required, and what the structure of a receipt is.

### CODE REVIEW

**Review by dmitrizagidulin — APPROVED** [review]
:+1:

**Comment by robinbryce** [comment]
Just to be clear, we very much appreciate the work on the cose merkle proof format and intend to use that by default where we can.

**Comment by robinbryce** [comment]
And also of course, LGTM

File: draft-ietf-scitt-architecture.md
Comment (OR13) [inline draft-ietf-scitt-architecture.md:959]: if we dare, we can make this an actual well known software supply chain media type

**Comment by OR13** [comment]
If you approved this PR, you should probably review and approve https://github.com/cose-wg/draft-ietf-cose-merkle-tree-proofs/pull/8

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:894]:
```suggestion
  2 => tstr; sub, the subject
```
So close :-)

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:903]: So this is a transitional version, where we have to decide what the difference between 13 and 393 is?

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:902]: Not sure how happy everybody going for "simplicity" will be with optional attributes in the protected header.

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:912]: a header map

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:898]: type name conflict with line 910. This one should probably be:

```suggestion
Signed_Statement_Protected_Header = {
```

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:911]: a CBOR data item including the protected header map. Probably should be:

```suggestion
    protected : bstr .cbor Signed_Statement_Protected_Header / bstr .size 0 
```

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:842]:
```suggestion

label = int / tstr
value = any

Inclusion_Proofs = [ + bstr ]
```

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:854]:
```suggestion
  * label => value
```

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:895]:
```suggestion
  * label => value
```

**Review by henkbirkholz — CHANGES_REQUESTED** [review]
I might have some mistakes, as I was lazily using the github review/suggestion feature.

File: draft-ietf-scitt-architecture.md
Comment (OR13) [reply draft-ietf-scitt-architecture.md:903]: Yep, not trying to fix everything at once.

File: draft-ietf-scitt-architecture.md
Comment (OR13) [reply draft-ietf-scitt-architecture.md:902]: I am fine banning x509 from headers... But it was my understanding that it was required to be supported, and it clearly can't be made mandatory.

File: draft-ietf-scitt-architecture.md
Comment (OR13) [reply draft-ietf-scitt-architecture.md:912]: Wut

File: draft-ietf-scitt-architecture.md
Comment (OR13) [reply draft-ietf-scitt-architecture.md:911]: Thank you!

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:916]: see unlabeled CDDL fragment at the end of this Section:

https://www.rfc-editor.org/rfc/rfc9052.html#name-header-parameters

Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:916]: Did I forget to include: `Signed_Statement_Unprotected_Header = { * label => value }` ?
That is what I meant. It can end up as: `unprotected: {}`

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:910]: Receipt_Unprotected_Header has a redundant definition. Probably needs even more qualification in the type name? Where does -222 go and where does -333 go?

File: draft-ietf-scitt-architecture.md
Comment (henkbirkholz) [inline draft-ietf-scitt-architecture.md:881]:
```suggestion
    protected : bstr .cbor Receipt_Protected_Header,
```

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:963]: yes, we should... what that type should be, that's another fun conversation.

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:907]: agreed, let's queue this up separately. @fournet was working on a new PR

**Comment by ad-l** [comment]
@OR13 I added the changes from the IETF Hackathon, including the new statement-specific registration information in the receipt and the description of transparent statement verification algorithm. The examples will have to be aligned to the new CDDL

**Comment by OR13** [comment]
These improvements seem excellent... I would approve them, if I could, but I opened the PR.

File: draft-ietf-scitt-architecture.md
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:928]:
```suggestion
Example transparent statement:
```

**Review by SteveLasker — APPROVED** [review]
LGTM

**Comment by SteveLasker** [comment]
@henkbirkholz , can you please re-review your "requested changes" review

**Review by ad-l — APPROVED** [review]
Even though some of the text needs to be adjusted to the receipt changes I think we should merge this as quickly as possible

File: draft-ietf-scitt-architecture.md
Comment (ad-l) [reply draft-ietf-scitt-architecture.md]: This should be addressed by my latest changes

File: draft-ietf-scitt-architecture.md
Comment (ad-l) [reply draft-ietf-scitt-architecture.md]: According to the current IANA registry https://www.iana.org/assignments/cose/cose.xhtml kccs is the header that has the correct type (map) and semantics (CWT claim set). So I changed the receipt type to use 14

File: draft-ietf-scitt-architecture.md
Comment (ad-l) [reply draft-ietf-scitt-architecture.md]: The semantics is that you need either a `kid` or a `x5chain`, but not both. There is no easy way to specify this in the syntax of the headers so I think this has to be specified in text

File: draft-ietf-scitt-architecture.md
Comment (ad-l) [reply draft-ietf-scitt-architecture.md:958]: Let's not block this PR on this. Picking one of the BOM types could bias the architecture and we want to be completely neutral

**Review by fournet — APPROVED** [review]
I also suggest we merge then work on the details.

**Comment by SteveLasker** [comment]
Merging as Henks comment was resolved

**Pull request merged**
